Vic2ToHoI4/Source/HOI4World/HoI4FocusTree.cpp

```diff
@@ -2063,6 +2063,283 @@ void HoI4FocusTree::addGPWarBranch(shared_ptr<HoI4::Country> Home,
 	nextFreeColumn += 2 * static_cast<int>(max(newAllies.size(), GCTargets.size()));
 }
 
+std::map<std::string, int> HoI4FocusTree::determineCoreHolders(std::shared_ptr<HoI4::Country> theCountry,
+	 const std::map<int, HoI4::State>& states
+){
+	std::map<std::string, int> coreHolders;
+
+	for (const auto& coreState: theCountry->getCoreStates())
+	{
+		if (const auto& owner = states.find(coreState)->second.getOwner(); owner != theCountry->getTag())
```

Comment:
While in practice this isn't going to run into non-existent states, that requires knowledge of the greater context. Put in checks against states.end() and second being nullptr so that this is obviously correct.

Reply:
As an advantage you'll have the state cached for later use, simplifying some of the uses.

Reply:
>Put in checks against states.end() and second being nullptr so that this is obviously correct.

Interestingly, second is HoI4::State and not a pointer - from `const std::map<int, HoI4::State>& states`. I'm not sure how do I check if it is valid.

Reply:
Ah, my mistake. In that case it can never be invalid, so no need for a check!